src/domain/in-app-notification-reader/dto/in.app.notification.user.mentioned.ts

```diff
@@ -0,0 +1,29 @@
+import { ObjectType } from '@nestjs/graphql';
+import {
+  InAppNotificationContributorMentionedPayload,
+  NotificationEventType,
+  InAppNotificationCategory,
+} from '@alkemio/notifications-lib';
+import { CommunityContributorType } from '@common/enums/community.contributor.type';
+import { IContributor } from '@domain/community/contributor/contributor.interface';
+import { InAppNotification } from '../in.app.notification.interface';
+import { InAppNotificationState } from '@domain/in-app-notification/in.app.notification.state';
+
+@ObjectType('InAppNotificationUserMentioned', {
+  implements: () => InAppNotification,
+})
+export class InAppNotificationUserMentioned implements InAppNotification {
+  type!: NotificationEventType.COMMUNICATION_USER_MENTION;
+  // fields resolved by a concrete resolver
+  contributorType?: CommunityContributorType;
+  comment?: string;
+  commentUrl?: string;
+  // inherited, resolved by the interface resolvers
+  id!: string;
+  category!: InAppNotificationCategory;
+  receiver!: IContributor;
+  triggeredBy?: IContributor;
+  state!: InAppNotificationState;
+  triggeredAt!: Date;
```

Comment:
I believe apart from the base abstract class, it will be good to have 3 classes derived from it for your notification types, there will be a lot of repetition otherwise.